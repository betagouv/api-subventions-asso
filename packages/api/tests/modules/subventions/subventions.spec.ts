import request from "supertest"
import getUserToken from "../../__helpers__/getUserToken";
import osirisRequestRepository from '../../../src/modules/providers/osiris/repositories/osiris.request.repository';
import fonjepRepository from '../../../src/modules/providers/fonjep/repositories/fonjep.repository';
import FonjepEntityFixture from '../providers/fonjep/__fixtures__/entity';
import OsirisRequestEntityFixture from '../providers/osiris/__fixtures__/entity';
import { Document } from 'mongodb';

const g = global as unknown as { app: unknown }

let osirisDocumentId: string | undefined;
let fonjepDocumentId: string | undefined;

describe("/subvention", () => {
    beforeEach( async () => {
        osirisDocumentId = ((await osirisRequestRepository.add(OsirisRequestEntityFixture)) as Document)._id.toString();
        fonjepDocumentId = ((await fonjepRepository.create(FonjepEntityFixture)) as Document)._id.toString();
    })
    
    describe("/{id}", () => {
        it("should return a subvention from osiris-request collection", async () => {
            const response = await request(g.app)
                .get(`/subvention/${osirisDocumentId}`)
                .set("x-access-token", await getUserToken())
                .set('Accept', 'application/json');
            expect(response.statusCode).toBe(200);
            expect(response.body).toMatchSnapshot();
        });
        it("should return a subvention from fonjep collection", async () => {
            const response = await request(g.app)
                .get(`/subvention/${fonjepDocumentId}`)
                .set("x-access-token", await getUserToken())
                .set('Accept', 'application/json');
            expect(response.statusCode).toBe(200);
            expect(response.body).toMatchSnapshot();
        })
    })
})