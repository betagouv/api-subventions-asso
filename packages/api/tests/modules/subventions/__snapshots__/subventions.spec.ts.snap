// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/subvention /{id} should return a subvention from fonjep collection 1`] = `
Object {
  "subvention": Object {
    "annee_demande": Object {
      "last_update": "2022-01-02T00:00:00.000Z",
      "provider": "Fonjep",
      "type": "number",
      "value": 2022,
    },
    "date_fin": Object {
      "last_update": "2022-01-02T00:00:00.000Z",
      "provider": "Fonjep",
      "type": "object",
      "value": "2022-01-03T00:00:00.000Z",
    },
    "montants": Object {
      "accorde": Object {
        "last_update": "2022-01-02T00:00:00.000Z",
        "provider": "Fonjep",
        "type": "number",
        "value": 500,
      },
    },
    "service_instructeur": Object {
      "last_update": "2022-01-02T00:00:00.000Z",
      "provider": "Fonjep",
      "type": "string",
      "value": "XXX",
    },
    "siret": Object {
      "last_update": "2022-01-02T00:00:00.000Z",
      "provider": "Fonjep",
      "type": "string",
      "value": "00000000000002",
    },
    "status": Object {
      "last_update": "2022-01-02T00:00:00.000Z",
      "provider": "Fonjep",
      "type": "string",
      "value": "En cours",
    },
  },
  "success": true,
}
`;

exports[`/subvention /{id} should return a subvention from osiris-request collection 1`] = `
Object {
  "subvention": Object {
    "actions_proposee": Array [],
    "contact": Object {
      "email": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
    },
    "date_commision": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "object",
      "value": "2022-01-01T00:00:00.000Z",
    },
    "dispositif": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "undefined",
    },
    "montants": Object {
      "accorde": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
      "demande": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
      "propose": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
      "total": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
    },
    "pluriannualite": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "undefined",
    },
    "service_instructeur": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "undefined",
    },
    "siret": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "string",
      "value": "00000000000001",
    },
    "sous_dispositif": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "undefined",
    },
    "status": Object {
      "last_update": "2022-01-01T00:00:00.000Z",
      "provider": "Osiris",
      "type": "undefined",
    },
    "territoires": Array [],
    "versement": Object {
      "acompte": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
      "compensation": Object {
        "n-1": Object {
          "last_update": "2022-01-01T00:00:00.000Z",
          "provider": "Osiris",
          "type": "undefined",
        },
        "reversement": Object {
          "last_update": "2022-01-01T00:00:00.000Z",
          "provider": "Osiris",
          "type": "undefined",
        },
      },
      "realise": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
      "solde": Object {
        "last_update": "2022-01-01T00:00:00.000Z",
        "provider": "Osiris",
        "type": "undefined",
      },
    },
  },
  "success": true,
}
`;
